AGENTS.md

```diff
@@ -0,0 +1,547 @@
+   - Follow the naming convention: `scaleway_{service}_{resource_name}`
+   - Ensure the resource function is properly imported
+
+8. **Add documentation**
+   - Create a new Go template file (`.tmpl`) in the appropriate subdirectory of `templates/` (e.g., `templates/resources/` for resources)
+   - Ensure examples are provided in the template
+   - Document all parameters and attributes in the template
+   - Run `make docs` to generate the documentation from the templates
+
+9. **Run validation**
+   - Execute `make build` to ensure the code compiles successfully
+   - Execute `make lint` to ensure the code passes all linter checks
+   - Execute `make testacc` to verify acceptance tests pass
+   - Execute `make format` to ensure proper code formatting
+
+## Testing Guidelines
+
+This section outlines the testing practices and guidelines for the Scaleway Terraform provider. Additional details can be found in the TESTING.md file, which should be followed when implementing tests.
+
+### Unit Testing
+
+- Follow the guidance in the official Terraform documentation for unit testing: [Unit Testing](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/unit-testing)
+- Unit tests should focus on logic and validation without external dependencies
+- Do not use mocking in unit tests - rely on pure function testing
+
+### Acceptance Testing
+
+- Only use acceptance testing with VCR cassettes to record actual API interactions
+- Do not use mocking - cassettes provide more reliable and accurate testing
+- Acceptance tests for a service should be in a dedicated package named after the service with `_test` suffix
+- Follow the standards in the official documentation: [Acceptance Tests](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests)
+
+### Test Structure and Validation
+
+- In test steps, verify important attributes of the schema as described in: [TestStep](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep)
+- Include steps that modify configuration to verify the update functionality works properly
+- Test both creation and subsequent updates in the same test flow
+- Ensure tests are isolated and can run in parallel
+
+## Cassette Management Guardrails
+
+VCR cassettes (recorded API interactions) are essential for reliable acceptance testing but can become very large. To prevent context overload:
+
+### Best Practices
+
+1. **Never Manually Modify Cassettes**
+   - Cassettes are automatically recorded by acceptance tests and should never be manually modified
+   - Always run tests with `TF_UPDATE_CASSETTES=true` to properly record new interactions
+   - Manual modifications can break test consistency and lead to unreliable tests
+
+2. **Cassette Compression**
+
+1. **Cassette Compression**
+   - Always compress cassettes using the vcr-compressor tool:
+   ```bash
+   go run ./cmd/vcr-compressor internal/services/{service}/testdata/{resource}.cassette.yaml
```

Comment:
As of today the command needs to run without the `.yaml` extension, so either this line should change to reflect this, or we should allow vcr-compressor to be launched with the full name of the file, as we often use the shell auto-completion to write the argument.